crates/noirc_evaluator/src/brillig/brillig_ir.rs

```diff
@@ -303,13 +324,6 @@ impl BrilligContext {
     /// the VM.
     pub(crate) fn return_instruction(&mut self, return_registers: &[RegisterIndex]) {
         for (destination_index, return_register) in return_registers.iter().enumerate() {
-            // If the destination register index is more than the latest register,
```

Comment:
Rationale is, why do we think this would ever happen? destination_index starts from 0..return_register, but we know we have index space for at least that many registers

Reply:
I suppose I could have an ensure_allocated method if it makes analysis easier

Reply:
Thats a good point -- I think this was for the case that you have only initialized 1 register due to the input to the function being one Field, but the output is lets say 5 Fields. The latest register would be 1 because we initialized a register for the input, but the destination would require 5 registers to store the function's return

Reply:
Yeah I started to have similar doubts - just coded an ensure_register_is_allocated, I'll push it